pennylane/pauli/utils.py

```diff
@@ -92,20 +93,96 @@ def is_pauli_word(observable):
+def _is_pw_pauli(observable: Union[PauliX, PauliY, PauliZ, Identity]):
+    return True
+
+
+@is_pauli_word.register
+def _is_pw_tensor(observable: Tensor):
     pauli_word_names = ["Identity", "PauliX", "PauliY", "PauliZ"]
-    if isinstance(observable, Tensor):
-        return set(observable.name).issubset(pauli_word_names)
+    return set(observable.name).issubset(pauli_word_names)
+
+
+@is_pauli_word.register
+def _is_pw_ham(observable: Hamiltonian):
+    return False if len(observable.ops) != 1 else is_pauli_word(observable.ops[0])
+
+
+@is_pauli_word.register
+def _is_pw_prod(observable: Prod):
+    return all(is_pauli_word(op) for op in observable)
+
+
+@is_pauli_word.register
+def _is_pw_sprod(observable: SProd):
+    return is_pauli_word(observable.base)
+
+
+@singledispatch
+def pauli_word_prefactor(observable):
+    """If the operator provided is a valid Pauli word (i.e a single term which may be a tensor product
+    of pauli operators), then this function extracts the prefactor.
+
+    Args:
+        observable (~.Operator): the operator to be examined
+
+    Returns:
+        Union[int, float, complex]: The scaling/phase coefficient of the Pauliword.
+
+    Raises:
+        ValueError: If an operator is provided that is not a valid PauliWord.
+
+    **Example**
+
+    >>> pauli_word_prefactor(qml.Identity(0))
+    1
+    >>> pauli_word_prefactor(qml.PauliX(0) @ qml.PauliY(1))
+    1
+    >>> pauli_word_prefactor(qml.PauliX(0) @ qml.PauliY(0))
+    1j
+    """
+    raise ValueError(f"Expected a valid PauliWord, got {observable}")
+
+
+@pauli_word_prefactor.register(PauliX)
+@pauli_word_prefactor.register(PauliY)
+@pauli_word_prefactor.register(PauliZ)
+@pauli_word_prefactor.register(Identity)
+def _pw_prefactor_pauli(observable: Union[PauliX, PauliY, PauliZ, Identity]):
+    return 1
+
+
+@pauli_word_prefactor.register
+def _pw_prefactor_tensor(observable: Tensor):
+    if is_pauli_word(observable):
+        from .conversion import pauli_sentence
+
+        return list(pauli_sentence(observable).values())[0]  # only one term,
+    raise ValueError(f"Expected a valid PauliWord, got {observable}")
+
 
-    if isinstance(observable, Hamiltonian):
-        return False if len(observable.ops) > 1 else is_pauli_word(observable.ops[0])
+@pauli_word_prefactor.register
+def _pw_prefactor_ham(observable: Hamiltonian):
+    if is_pauli_word(observable):
+        return observable.coeffs[0]
+    raise ValueError(f"Expected a valid PauliWord, got {observable}")
 
-    if isinstance(observable, Prod):
-        return all(is_pauli_word(op) for op in observable)
 
-    if isinstance(observable, SProd):
-        return is_pauli_word(observable.base)
+@pauli_word_prefactor.register(Prod)
+@pauli_word_prefactor.register(SProd)
+def _pw_prefactor_prod_sprod(observable: Union[Prod, SProd]):
+    ps = observable._pauli_rep  # pylint:disable=protected-access
+    if ps is not None and len(ps) == 1:
+        return list(ps.values())[0]
```

Comment:
Especially if we are making `Operator.pauli_rep` public, is it worth actually having this utility?  Or can we just "recommend" doing `list(ps.values())[0]`? Are we making the interface more complicated than the code it obscures?

Reply:
I think this is a fair question. `is_pauli_word` also could be replaced by `len(op.pauli_rep)==1` , right? :thinking:

Reply:
Its definitely not the cleanest interface to ask a user to do `list(ps.values())[0]`, but I think having a specialized class attribute to track the pre-factor is also overkill. I don't want this class to end up ballooning to the point where it starts affecting the performance, because that was the whole point of creating it in the first place. 

I think if its as simple as `len(op.pauli_rep) == 1` then we can just expect the user to use that, but for slightly more complicated functions (like `list(ps.values())[0]`) I would prefer having a function (`qml.pauli_prefact(ps)`).